File: 1. Creational Design Patterns/Factory Method/Java/FactoryMethodTemplate.java
// Factory Method Java Template
interface Product {}
abstract class Creator { abstract Product factoryMethod(); }
class ConcreteProduct implements Product {}
class ConcreteCreator extends Creator { Product factoryMethod(){ return new ConcreteProduct(); } }
public class FactoryMethodTemplate { public static void main(String[] a){ Creator c = new ConcreteCreator(); c.factoryMethod(); } }

File: 1. Creational Design Patterns/Factory Method/Go/FactoryMethodTemplate.go
// Factory Method – Go Template
package main

type Product interface{}

type Creator interface{ FactoryMethod() Product }

type ConcreteCreator struct{}
func (ConcreteCreator) FactoryMethod() Product { return struct{}{} }

func main(){ var c Creator = ConcreteCreator{}; _ = c.FactoryMethod() }
